modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/assignments/Assignments.js

```diff
@@ -32,15 +40,34 @@ const assignmentSectionComponents = {
 
 const Assignments = (props) => {
 	const {selectedItem} = useContext(DiagramBuilderContext);
+	const {accountEntryId} = useContext(DefinitionBuilderContext);
 	const assignments = selectedItem?.data?.assignments;
 
 	const assignmentType = getAssignmentType(assignments);
 
+	const [accountRoles, setAccountRoles] = useState([]);
+	const [networkStatus, setNetworkStatus] = useState(4);
 	const [section, setSection] = useState(assignmentType || 'assetCreator');
 	const [sections, setSections] = useState([{identifier: `${Date.now()}-0`}]);
 
 	const AssignmentSectionComponent = assignmentSectionComponents[section];
 
+	const {resource} = useResource({
+		fetchOptions: {
+			headers: {
+				...headers,
+				'accept': `application/json`,
+				'x-csrf-token': Liferay.authToken,
+			},
+		},
+		fetchPolicy: 'cache-first',
+		link: `${window.location.origin}${contextUrl}${userBaseURL}/roles`,
+		onNetworkStatusChange: setNetworkStatus,
+		variables: {
+			pageSize: -1,
+		},
+	});
+
```

Comment:
Hello, @naraandrade. I was searching for the use of `useResource`  and in some places it's being used the `import {fetch} from 'frontend-js-web';`
as in this file `/data_layout_builder/js/components/rules/editor/Actions.es.js`. Maybe using it the code could be smaller and avoid some default values. But fell free to analyse and make some changes in the future.

Reply:
Thanks for pointing that out, @alinedoleron. We are using the `fetch` from `frontend-js-web` in many files in this module too. Unfortunately, this is not a replacement we can do very fast and I would like to avoid introducing bugs. I'll open a technical debt for it and we'll work on this on the future. Thanks :)

Reply:
Technical debt created: https://issues.liferay.com/browse/LPS-174808.